logging/auto_roll_logger.cc

```diff
@@ -278,13 +278,19 @@ Status CreateLoggerFromOptions(const std::string& dbname,
       InfoLogFileName(dbname, db_absolute_path, options.db_log_dir);
 
   const auto& clock = env->GetSystemClock();
-  // In case it does not exist
-  s = env->CreateDirIfMissing(dbname);
-  if (!s.ok()) {
-    return s;
-  }
-
-  if (!options.db_log_dir.empty()) {
+  if (options.db_log_dir.empty()) {
+    // In case it does not exist.
+    s = env->CreateDirIfMissing(dbname);
+    if (!s.ok()) {
+      return s;
+    }
+  } else {
+    // Ignore the error returned during creation of dbname because dbname and
+    // db_log_dir can be on different filesystems in which case dbname will not
+    // exist and error should be ignored.
+    // db_log_dir creation will handle the error in case there is any error in
+    // the creation of dbname on same filesystem.
+    env->CreateDirIfMissing(dbname).PermitUncheckedError();
```

Comment:
How about
```cpp
s = env->CreateDirIfMissing(dbname);
if (!s.ok()) {
  if (options.db_log_dir.empty()) {
    return s;
  } else {
     s = Status::OK();
  }
}

assert(s.ok());

if (!options.db_log_dir.empty()) {
```

Reply:
I actually wonder if we can just evaluate the fname for the LOG file, and then extract its parent directory path. Next, we create a directory for this path. Not sure how many of our tests/users rely on a `dbname` being created here.

Reply:
I can try with the tests and not sure about the users. Since this a patch fix do we want to change the behavior? Or address that in separate PR?

Reply:
Let's do it in another PR